ui/app/models/database/role.js

```diff
@@ -4,100 +4,164 @@
  */
 
 import Model, { attr } from '@ember-data/model';
-import { computed } from '@ember/object';
-import { alias } from '@ember/object/computed';
 import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
-import { expandAttributeMeta } from 'vault/utils/field-to-attrs';
 import { getRoleFields } from 'vault/utils/model-helpers/database-helpers';
+import { expandAttributeMeta } from 'vault/utils/field-to-attrs';
+import { withModelValidations } from 'vault/decorators/model-validations';
+
+const validations = {
+  database: [
+    {
+      validator(model) {
+        const { database } = model;
+        if (database) return true;
+      },
+      message: 'Database is required',
+    },
```

Comment:
There is a [built in validator for presence](https://github.com/hashicorp/vault/blob/main/ui/app/utils/model-helpers/validators.js#L15).
```suggestion
    { type: 'presence', message: 'Database is required' },
```